testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/security/authentication/deployment/SecuredEJBServlet.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015, Red Hat, Inc. and/or its affiliates, and individual
+ * contributors by the @authors tag. See the copyright.txt in the
+ * distribution for a full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.as.test.integration.web.security.authentication.deployment;
+
+import javax.ejb.EJB;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.HttpConstraint;
+import javax.servlet.annotation.ServletSecurity;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * A simple secured Servlet which calls a secured EJB. Upon successful authentication and authorization the EJB will return the
+ * principal's name. Servlet security is implemented using annotations.
+ *
+ * @author Sherif Makary
+ *
+ */
+@SuppressWarnings("serial")
+@WebServlet("/SecuredEJBServlet")
+@ServletSecurity(@HttpConstraint(rolesAllowed = "guest"))
+public class SecuredEJBServlet extends HttpServlet {
+
+    @EJB
+    private SecuredEJB securedEJB;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```

Comment:
This servlet should call the EJB and return some kind of response so that the client can verify that it is actually working.